### PR TITLE
Adding a Cache to Transaction Receipts Store

### PR DESCRIPTION
**Pull request by ajlopezrsk**

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/RskContext.java [hunk 1]
```diff
@@ -835,8 +835,8 @@ protected NodeRunner buildNodeRunner() {
                 getRskSystemProperties().flushNumberOfBlocks(),
                 getCompositeEthereumListener(),
                 getTrieStore(),
-                getBlockStore()
-        ));
+                getBlockStore(),
+                null));
```
Comment (patogallaiovlabs) [inline rskj-core/src/main/java/co/rsk/RskContext.java:839]: Shouldn't call to "getReceiptStore()" instead of null?

**Comment by patogallaiovlabs** [comment]
As i see, it also fix the issue of flushing before stoping the node. The one that someone complained, that it was not flushing because of never reaching the minimum flushing amount of blocks before stoping it. Am i right?

File: rskj-core/src/main/java/co/rsk/RskContext.java (hunk 1 above)
Comment (ajlopezrsk) [reply rskj-core/src/main/java/co/rsk/RskContext.java]: Good catch! Fixing it

**Comment by ajlopezrsk** [comment]
> As i see, it also fix the issue of flushing before stoping the node. The one that someone complained, that it was not flushing because of never reaching the minimum flushing amount of blocks before stoping it. Am i right?

Yes, we could fix now, or later. Maybe add a QA explicit test

**Pull request merged**